ansys/mapdl/core/mapdl_grpc.py

```diff
@@ -1361,8 +1362,127 @@ def _get(self, entity, entnum, item1, it1num, item2, it2num):
 
         raise RuntimeError(f"Unsupported type {getresponse.type} response from MAPDL")
 
+    def download(self,
+                 files=None,
+                 chunk_size=DEFAULT_CHUNKSIZE,
+                 progress_bar=True): # pragma: no cover
+        """Download files from the gRPC instance workind directory
+
+        Parameters
+        ----------
+        files : str or List[str] or Tuple(str)
+            Name of the file on the server. File must be in the same
+            directory as the mapdl instance.  List current files with
+            :func:`Mapdl.directory <ansys.mapdl.core.Mapdl.list_files>`.
+
+            Alternatively, you can also specify the following options:
+
+            * `'ALL'`
+                Download all the files ending in: 'out', 'full', 'rst', 'cdb',
+                'err', 'db', or 'log', in the MAPDL working directory
+                (:func:`Mapdl.directory <ansys.mapdl.core.Mapdl.directory>`).
+
+            * `'EVERYTHING'`
+                Download every single file in the MAPDL working directory.
+
+            * `'file*'`
+                Global expressions can be used to match file names.
+
+            * List or tuple containing the file names.
+
+        chunk_size : int, optional
+            Chunk size in bytes.  Must be less than 4MB.  Defaults to 256 kB.
+
+        progress_bar : bool, optional Display a progress bar using
+            ``tqdm`` when ``True``.  Helpful for showing download
+            progress.
+
+        .. warning::
+            This feature is only available for MAPDL 2021R1 or newer.
+
+        .. note::
+
+            * The global pattern search does not search recursively.
+            * If you are in local and provide a file path, downloading files
+              from a different folder is allowed.
+              However it is not a recommended approach.
+
+        Examples
+        --------
+        Download all the simulation files ('out', 'full', 'rst', 'cdb', 'err', 'db', or 'log'):
+
+        >>> mapdl.download('all')
+
+        Download every single file in the MAPDL workind directory:
+
+        >>> mapdl.download('everything')
+
+        Download a single file:
+
+        >>> mapdl.download('file.out')
+
+        Download all the files starting with `'file'`:
+
+        >>> mapdl.download('file*')
+
+        """
+
+        extensions_all = ['OUT', 'FULL', 'RST', 'CDB', 'ERR', 'DB', 'LOG']
```

Comment:
@germa89 From my remembers when I was a solver developer, there are more file extensions existing for mapdl as : .rsdp .rfrq .mode .rth .rmg .dump .....

Reply:
I agree, but those are the files extension that will be downloaded when issuing `mapdl.download('all')`. I believe most of the time, the regular user is interested mainly in ['OUT', 'FULL', 'RST', 'CDB', 'ERR', 'DB', 'LOG'].

Reply:
We need a `download_project`. Prototype:
```
def download_project(self, extensions=['):